aiozmq/interface.py

```diff
@@ -194,3 +215,36 @@ def msg_received(self, data):
 
         data is the multipart tuple of bytes with at least one item.
         """
+
+MonitorEvents = {zmq.EVENT_ACCEPTED: 'accepted',
+                 zmq.EVENT_ACCEPT_FAILED: 'accept failed',
+                 zmq.EVENT_BIND_FAILED: 'bind failed',
+                 zmq.EVENT_CLOSE_FAILED: 'close failed',
+                 zmq.EVENT_CLOSED: 'closed',
+                 zmq.EVENT_CONNECTED: 'connected',
+                 zmq.EVENT_CONNECT_DELAYED: 'connect delayed',
+                 zmq.EVENT_CONNECT_RETRIED: 'connect retried',
+                 zmq.EVENT_DISCONNECTED: 'disconnected',
+                 zmq.EVENT_LISTENING: 'listening',
+                 zmq.EVENT_MONITOR_STOPPED: 'monitor stopped'}
+
+
+class ZmqEventProtocol(ZmqProtocol):
```

Comment:
When socket event monitoring is enabled on a socket the ZMQ internals binds a zmq.PAIR to an inproc:// endpoint. Another zmq.PAIR is created, connected to the inrpoc:// endpoint and then returned to the user. 

The socket monitor needs its own protocol so we can intercept the messages in the standard msg_received callback, perform any necessary processing (parse_monitor_message) to massage the event into something useful then return it to the user. If the socket event monitor did not have its own protocol we would need to inspect every message through msg_received to check if it is a socket event kind of message. I do not think this would be an efficient approach. In any case this would result in an unconventional use of protocols... it would result in connection_made being called twice as the protocol used on the main socket gets passed in to the aiozmq.create_zmq_connection call when creating the monitor transport and protocol. This would not be intuitive and harder to maintain.
 
It might be possible to internally hide the event protocol used by the monitor socket and wire it up to pass any received events over the the user's ZmqProtocol.event_received method. This would simplify the API exposed to a user but would add complexity (e.g. chaining two protocols together) to the internals of aiozmq. I'll try to rework the implementation more in line with your suggestions.